Sources/KlaviyoSwift/APIRequestErrorHandling.swift

```diff
@@ -12,12 +12,48 @@ struct ErrorHandlingConstants {
     static let maxBackoff = 60 * 3 // 3 minutes
 }
 
+enum InvalidField: Equatable {
+    case email
+    case phone
+
+    /// gets the invalid field based on the source.pointer from klaviyo API.
+    /// this assumes that source.pointer will not change
+    /// Client APIs to have better error codes in the future at which point we should use that instead of source.pointer
+    /// - Parameter sourcePointer: pointers to the source of the error
+    /// - Returns: the field that is invalid else `nil`
+    static func getInvalidField(sourcePointer: String) -> InvalidField? {
+        if sourcePointer == "/data/attributes/phone_number" {
+            return .phone
+        }
+        if sourcePointer == "/data/attributes/email" {
+            return .email
+        }
+
+        return nil
+    }
+}
+
 private func getDelaySeconds(for count: Int) -> Int {
     let delay = Int(pow(2.0, Double(count)))
     let jitter = environment.randomInt()
     return min(delay + jitter, ErrorHandlingConstants.maxBackoff)
 }
 
+private func parseError(_ data: Data) -> [InvalidField]? {
+    var invalidFields: [InvalidField]?
+    do {
+        let errorResponse = try JSONDecoder().decode(ErrorResponse.self, from: data)
```

Comment:
There is a decode method (I think) in the environment which might aid you in testing.

Reply:
Ah good call. Using the one from environment and it helped simplify the code a bit as well.